crates/nu-engine/src/documentation.rs

```diff
@@ -318,6 +323,97 @@ fn get_documentation(
     }
 }
 
+fn get_ansi_color_for_component_or_default(
+    engine_state: &EngineState,
+    theme_component: &str,
+    default: &str,
+) -> String {
+    if let Some(color) = &engine_state.get_config().color_config.get(theme_component) {
+        let mut caller_stack = Stack::new();
+        let span = Span::unknown();
+
+        let argument = get_argument_for_color_value(&engine_state, default, color, span);
+
+        // Call ansi command using argument
+        if let Some(decl_id) = engine_state.find_decl(b"ansi", &[]) {
+            if let Ok(result) = eval_call(
+                engine_state,
+                &mut caller_stack,
+                &Call {
+                    decl_id,
+                    head: span,
+                    arguments: vec![argument],
+                    redirect_stdout: true,
+                    redirect_stderr: true,
+                    parser_info: HashMap::new(),
+                },
+                PipelineData::Empty,
+            ) {
+                if let Ok((str, ..)) = result.collect_string_strict(span) {
+                    return str;
+                }
+            }
+        }
+    }
+
+    default.to_string()
+}
+
+fn get_argument_for_color_value(
+    engine_state: &&EngineState,
```

Comment:
This should probably remain a simple `&` and if necessary remove a level through dereference at the call site.
```suggestion
    engine_state: &EngineState,
```